tools/CleanOutput.py

```diff
@@ -0,0 +1,77 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import os
+import re
+import logging
+
+
+def clean_output(input_file, output_dir):
+    """
+    Deletes all output files generated by a run with the `input_file` in
+    the `output_dir`.
+
+    The `input_file` must list its expected output files in a comment:
+    ```yaml
+    # ExpectedOutput:
+    #   Volume0.h5
+    #   Reduction.h5
+    ```
+    This function raises an error if the expected output files were not found.
+    """
+    found_indentation = None
+    for line in open(input_file, 'r'):
+        # Iterate through the file until we find the `ExpectedOutput` comment
+        if found_indentation is None:
+            matched_indentation = re.match('#([ ]*)ExpectedOutput:', line)
+            if matched_indentation is not None:
+                found_indentation = matched_indentation.groups()[0] + '  '
+        else:
+            # Now collect the output files listed in the comment
```

Comment:
I think it would be useful to add some detail to this comment by adding a sentence similar to this: "Look for lines with two spaces more indentation than the preceding `ExpectedOutput` comment". Feel free to disagree and ignore.